File: backend/parking-api/src/main/java/com/chatchatabc/api/domain/service/ParkingLotService.java
package com.chatchatabc.api.domain.service;

import com.chatchatabc.api.application.dto.parking_lot.ParkingLotDTO;

public interface ParkingLotService {
    ParkingLotDTO registerParkingLot(
            String ownerId,
            String name,
            Double latitude,
            Double longitude,
            String address,
            String description,
            Integer capacity
    );
}

File: backend/parking-service/src/main/kotlin/com/chatchatabc/service/domain/service/ParkingLotService.kt
package com.chatchatabc.service.domain.service

import com.chatchatabc.service.domain.model.ParkingLot
import org.springframework.stereotype.Service

@Service
interface ParkingLotService {

    /**
     * Register a parking lot
     */
    fun register(ownerId: String, parkingLot: ParkingLot): ParkingLot

    /**
     * Update a parking lot
     */
    fun update(ownerId: String, parkingLotId: String, newParkingLotInfo: ParkingLot): ParkingLot
}